### PR TITLE
Added authorization template

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
**Release note**:

```release-note
Authorization adapter template.
Any authorization module needs to answer the question: Is a “subject” allowed to do an “action”. 
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1458)

### CODE REVIEW

File: template/authz/template.proto [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+message Template {
+  // known fields:
+  // - name: The name of the subject.
+  // - groups: Groups the subject belongs to.
+  map<string, istio.mixer.v1.config.descriptor.ValueType> subject = 1;
+
+  // known fields:
+  // - namespace: Namespace the target action is taking place in.
+  // - service: The Service the action is being taken on.
+  // - method: The what action is being taken.
```
Comment (liminw) [inline template/authz/template.proto]: method: HTTP method or gRPC method.

File: template/authz/template.proto [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  // - path: The resource within the service.
```
Comment (liminw) [inline template/authz/template.proto]: path: HTTP REST paths. It is ignored in gRPC case.

**Review by liminw — APPROVED** [review]

**Comment by mangchiandjjoe** [comment]
@geeknoid @douglas-reid @mandarjog @guptasu @ZackButcher @liminw 

Could you please review the PR?

**Comment by geeknoid** [comment]
I started a conversation about auth templates here: https://groups.google.com/forum/#!topic/istio-integrations/hWPkl95MmHM

Let's discuss there for a while.

Thanks.

**Comment by geeknoid** [comment]
This repo is no longer accepting PRs. Please resubmit this change to the istio/istio repo.

Thanks.

**Pull request closed without merging**